Fix encoding issue between Python 2 and 3

JrGoodle opened the pull request:
Addresses #130 

Fix for issue introduced in #121 

I tested this on my own project and it worked: https://circleci.com/gh/JrGoodle/clowder/154

It probably still needs to be tested with Python 2

chfast commented on codecov/__init__.py:
```diff
@@ -719,8 +719,16 @@ def main(*argv, **kwargs):
+                            basestring
+                        except NameError:
+                            basestring = str
+                        if isinstance(reports, basestring):
```
Almost good.

For Python 2 you want `if isinstance(reports, unicode)` (not `(str, unicode)`), 
for python 3 you want `if isinstance(reports, str)`.

So the correct fix would be:

```python
try:
    unicode
except NameError:
    # Must be Python 3, make an alias for str.
    unicode = str
if isinstance(reports, unicode):
    reports = reports.encode('utf-8')
```

However, Python 2 has `bytes` being an alias for `str`. So you can do:

```python
if not isinstance(reports, bytes):
    reports = reports.encode('utf-8')
```

JrGoodle commented:
@chfast I made the suggested change

@stevepeak I also configured the tests to skip invalid CI providers. I had to disable a number of extra tests for AppVeyor to pass (including the one specifically for AppVeyor). There was another string encoding bug exposed in the tests that I fixed. let me know if there are any changes you'd like for this part of the patch, but I think having fewer tests that consistently pass is better than having more tests that consistently fail

in addition, I updated the .gitignore from gitignore.io because I noticed it was trying to track a `.DS_Store` file

blueyed commented:
Regarding .gitignore, I suggest configuring a global gitignore file.
Might be good to add PyCharm related files here, but there seem to be many unneeded additions now.

JrGoodle commented:
I generated the gitignore using the following options on gitignore.io: linux,macos,python,pycharm+all

It seem like a lot, but they're pretty well maintained and thorough. Would you rather I used a subset?

blueyed commented:
@JrGoodle 
I think it makes sense for a separate PR - it should not block this fix.
And then, yes - there is no need to ignore e.g. CMake related files in this project.
And like I said, you can have a global/user gitignore file, too.

JrGoodle commented:
@blueyed I removed the .gitignore commit from this PR

stevepeak commented:
Thank you @JrGoodle @chfast @blueyed  for your help. 👍
